File: src/Reference.CSharp/GeoContext.cs
using Microsoft.EntityFrameworkCore;
// ReSharper disable UnusedAutoPropertyAccessor.Global

namespace Reference.CSharp
{
    /// <summary>
    /// Geo Context
    /// </summary>
    public class GeoContext : DbContext
    {
        private readonly DBServerType dbServerType;

        public GeoContext(DBServerType databaseType)
        {
            dbServerType = databaseType;
        }

        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
            switch (dbServerType)
            {
                case DBServerType.SqlServer:
                    optionsBuilder.UseSqlServer(Settings.DBConnectionString);
                    break;
                case DBServerType.Postgres:
                    optionsBuilder.UseNpgsql(Settings.PostgresConnectionString);
                    break;
                case DBServerType.MySql:
                    optionsBuilder.UseMySql(Settings.MySqlConnectionString);
                    break;
            }
        }

        public DbSet<GeoEntity> GeoData { get; set; }
    }
}

File: src/Reference.VB/GeoContext.vb
Imports Microsoft.EntityFrameworkCore

''' <summary>
''' Geo Context
''' </summary>
Public Class GeoContext
    Inherits DbContext

    Private ReadOnly dbServerType As DBServerType

    Public Sub New(databaseType As DBServerType)
        dbServerType = databaseType
    End Sub

    Protected Overrides Sub OnConfiguring(optionsBuilder As DbContextOptionsBuilder)
        Select Case dbServerType
            Case DBServerType.SqlServer
                optionsBuilder.UseSqlServer(Settings.DBConnectionString)
            Case DBServerType.Postgres
                optionsBuilder.UseNpgsql(Settings.PostgresConnectionString)
        End Select
    End Sub

    Public Property GeoData As DbSet(Of GeoEntity)
End Class
